modules/aurora-mysql/variables.tf

```diff
@@ -199,17 +199,29 @@ variable "primary_cluster_component" {
 }
 
 variable "allow_ingress_from_vpc_accounts" {
-  type        = any
+  type = list(object({
+    vpc         = optional(string)
```

Comment:
just thinking aloud here. I don't have any immediate objections. but realize that `vpc` will be mapped to `name`. That requires knowledge of how `context.tf` attributes are employed in a `remote-state` lookup. That said, I don't mind it if we think there aren't too many other lookups that could confuse the convention.